provisioning/containers/README.md

```diff
@@ -0,0 +1,20 @@
+# Creating and uploading Docker-based graphical applications in CrownLabs
+
+This folder contains the files required to create docker-based graphical applications in CrownLabs, as an alternative solution to VMs.
+
+In order to simplify the process of creating such GUI applications, two containers are envisioned, running according to the *sidecar* model in Kubernetes:
+
+- **Common components**: a first container, under [gui-common](gui-common/), providing all the components required to provide a virtual desktop accessible from a web browser (i.e., virtual desktop environment, noVNC server, etc).
+- **Actual application**: a second container keeping the actual application, running in the above virtual desktop. Currently, the [pycharm](pycharm/) folder is an example of such an application, based on the PyCharm integrated development environment for Python.
+
+This architecture decouples the application itself, e.g. PyCharm, from the virtual desktop logical layer provided by _gui-common_.
+
+## Starting the application
+The application can be started by running both containers in the same pod.
+
+## Creating a new application
+A new application can be created by just *containerizing* it, i.e., writing a self-contained Dockerfile for the application layer, without introducing strict bindings with the desktop environments that would make the deployment monolithic and thus less maintainable.
+In other words, we can deploy new applications by replacing the PyCharm's Dockerfile with a new one, while keeping what is inside _gui-common_.
+
+Note that the `DISPLAY` environment variable must be the same used within the running `X` server instance and that both Xvnc and the application container must share the `/tmp/.X11-unix` directory that will contain the socket file used to communicate with the X server.
```

Comment:
I am not sure the message is so clear here. Can you please try to rephrase, or at least comment here, so that we can improve the readability of the sentences?

Reply:
The note on lines 19 and 20 is just for saying that who will make the application Docker image might want to be aware of those two "parameters" (the `DISPLAY` variable and the folder that will be mounted on `/tmp`), that will be set by the CrownLabs operator when issuing the instance of such application, but actually I guess an application could work perfectly without knowing about that, so the note could also be removed... It's more of a warning, just in case